pkg/cmd/cli/secrets/known_secret_types.go

```diff
@@ -12,13 +12,11 @@ type KnownSecretType struct {
 	RequiredContents util.StringSet
 }
 
-func (ks KnownSecretType) Matches(secretContent map[string][]byte) bool {
+func (ks KnownSecretType) Matches(secretContent []string) bool {
 	if secretContent == nil {
 		return false
 	}
-
-	secretKeys := util.KeySet(reflect.ValueOf(secretContent))
-	return reflect.DeepEqual(ks.RequiredContents.List(), secretKeys.List())
+	return reflect.DeepEqual(ks.RequiredContents.List(), secretContent)
```

Comment:
`KnownSecretTypes` operates off of `kapi.DockerConfigKey = ".dockercfg"` right now, which the `DeepEquals` just checks for equality against, so is that not the file ~~name~~ extension - which we lose knowledge of when users name their keys?